### ISSUE TITLE
sql: Record the correct name for a new index

### ISSUE THREAD

**Issue by maxlang** [cockroachdb/cockroach#13144]
When you create an index without specifying in a name, it gets assigned a name automatically:

```sql
root@:26257> CREATE INDEX ON test.test (id);
CREATE INDEX
root@:26257> SHOW CREATE TABLE test.test;
+-----------+-----------------------------------+
|   Table   |            CreateTable            |
+-----------+-----------------------------------+
| test.test | CREATE TABLE "test.test" (␤       |
|           |     id INT NULL,␤                 |
|           |     INDEX test_id_idx (id),␤      |
|           |     FAMILY "primary" (id, rowid)␤ |
|           | )                                 |
+-----------+-----------------------------------+
```

However in the event the name for this index is recorded as the empty string:

```json
{ "TableName": "test", "IndexName": "\"\"", "Statement": "CREATE INDEX ON test.test (id)", "User": "root", "MutationID": 5 }
```

We should either correctly record the automatically generated index name here or, if that's not possible, indicate it in the `finish_schema_change` event.

**Comment by dianasaur323** [comment]
@mrtracy Do you mind checking if this is still an issue? If not, let's close.

**Comment by dianasaur323** [comment]
Moving to 1.2

**Comment by knz** [comment]
This was improperly triaged as a web UI issue; it's actually an issue in the DDL logic.

cc @vivekmenezes for triage.

**Comment by vivekmenezes** [comment]
resolved by #22842
